test/test_energy_storage_device.py

```diff
@@ -28,24 +30,31 @@ def test_consistency_constant_evolve_functions(self):
                                dt)
         df.reset()
         P = 15.0  # watts/m2
+        error = 1e-2  # represents leniency of error caused
+                      # by dualfoil, not the code being tested
         df.evolve_one_time_step_constant_power(dt, P)
-        power = df.get_voltage() * df.get_current()
+        # Sign of current only indicates direction of flow
+        # No such thing as negative power; use abs val
+        power = abs(df.get_voltage() * df.get_current())
         self.assertAlmostEqual(df_manip.get_total_time(path),
                                dt)
-        self.assertAlmostEqual(power, P, delta=1e-2)
+        self.assertAlmostEqual(power, P, delta=error)
         df.reset()
         L = 0.7  # ohms-m2
         df.evolve_one_time_step_constant_load(dt, L)
-        load = df.get_voltage() / df.get_current()
+        # Sign of current only indicates direction of flow
+        # No such thing as negative load; use abs val
+        load = abs(df.get_voltage() / df.get_current())
         self.assertAlmostEqual(df_manip.get_total_time(path),
                                dt)
-        self.assertAlmostEqual(load, L, delta=1e-2)
+        self.assertAlmostEqual(load, L, delta=error)
         df.reset()
-        Vcut = 4.3  # cutoff voltage
-        C = -10.0   # current (discharge)
+        Vcut = 4.3    # cutoff voltage
+        C = -10.0     # current (discharge)
+        error = 1e-4  # this dualfoil simulation is more accurate
```

Comment:
OK I see you redefine `error` here. It means 0.1 mV of precision for voltage. Please elaborate.